### PR TITLE
Allow ForEach to For when there are comments 

### PR DESCRIPTION
**Pull request by Cosifne**
Related: [issue](26571)

Remove the check for comments for both VB and C# so the refactoring will be provided.
Change the corresponding tests.

### CODE REVIEW

**Comment by Cosifne** [comment]
@dotnet/roslyn-ide  Can someone approve this ?

File: src/EditorFeatures/CSharpTest/ConvertForEachToFor/ConvertForEachToForTests.cs [hunk 1]
```diff
@@ -372,11 +372,22 @@ void Method()
...
+    {
+        var array = new int[] { 1, 3, 4 };
+        for (int {|Rename:i|} = 0; i < array.Length; i++) ;
+    }
+}
+";
+            await TestInRegularAndScriptAsync(text, expected);
         }
 
+
```
Comment (mavasani) [inline src/EditorFeatures/CSharpTest/ConvertForEachToFor/ConvertForEachToForTests.cs]: Nit: unnecessary blank line.

File: src/Features/CSharp/Portable/ConvertForEachToFor/CSharpConvertForEachToForCodeRefactoringProvider.cs [hunk 2]
```diff
@@ -38,24 +36,6 @@ protected override ForEachStatementSyntax GetForEachStatement(TextSpan selection
                 return null;
             }
 
-            // check whether there is any comments between foreach and ) tokens
```
Comment (mavasani) [inline src/Features/CSharp/Portable/ConvertForEachToFor/CSharpConvertForEachToForCodeRefactoringProvider.cs]: Removing the bail out seems fine, but should we still be preserving trivia in the refactored code? My point is are we removing this check because some customer requested it when they did have such comment trivia in their code and didn't get to see this refactoring or just because Kasey noticed it and filed the issue being fixed here? If it is the former, then it is likely that the trivia in original code was something that the customer intended to be retained and we should not be dropping it.

Regardless, I feel there is not much _real_ code out there that has comments interspersed between the foreach loop header, so I am fine with both the original code and bail out approach OR your new approach. However, I would prefer that we don't make more similar changes in this area until we hear customer feedback or do a much bigger cleanup/refactoring to make all of our code refactorings consistent.

Comment (CyrusNajmabadi) [reply src/Features/CSharp/Portable/ConvertForEachToFor/CSharpConvertForEachToForCodeRefactoringProvider.cs]: Note: as per the dsicussion about this feature originally:

This feature is behaving *very* strangely.  Our other features do not do this.  i.e. we are *not* conservative with trivia in terms of having it block the feature.  Instead we do this:

1. we allow features with trivia, regardless of where it is.
2. we make a *best* effort to do a *good enough* job with *common* trivia patterns.  i.e. comments before the for/foreach statement would be something we'd care a lot aobut making work.  comments *inside* the header, not so much.
3. If a user runs into a case where we do a bad job with trivia, and their code pattern seems reasonable, we'll consider making a fix for them.
4. We will disable a feature due to trivia if it's a *reasonable* pattern *and* we do a very bad job *and* it's really hard to fix.

The last is basically the only time when we let trivia stop us.  This is rare, but can happen some times with things like pp-directives. (where we might *Really* break code, and where it can often be very tricky to get it right).

--

So, TLDR, i see this change as just making this feature behave like all our other features.  It was very odd when this was written (and there was a big discusison about all of this), and i'm happy to see this happen to keep things as consistent as possible.

Thanks!

**Comment by CyrusNajmabadi** [comment]
> I am fine with the current approach as well as the original approach. However, @jinujoseph @vatsalyaagrawal should decide if this meets the current 16.0 P4 bar as there is no actual customer requesting this change or reporting the current behavior as a bug.

Agreed.  This would not meet any bar for me.  What about moving to 16.1?
